dicom_utils/types.py

```diff
@@ -18,34 +18,134 @@
 WINDOW_WIDTH = 0x00281051
 UNKNOWN: Final = -1
 
+
 Dicom = Dataset
 DicomAttributeSequence = Sequence
 
 
+class ModalityError(Exception):
+    pass
+
+
+def get_value(dcm: Dicom, tag: Tag, default: Any) -> Any:
+    return dcm.get(tag).value if tag in dcm else default
+
+
 def get_tag_values(tags: Iterable[Tag], dcm: Dicom) -> Dict[Tag, Any]:
-    tags = {tag: dcm[tag].value if tag in dcm else None for tag in tags}
+    tags = {tag: get_value(dcm, tag, None) for tag in tags}
     return tags
 
 
-class SimpleImageType(Enum):
+# TODO: these would be better in dicom.py, but circular import issues
+def view_code_iterator(dcm: Dataset) -> Iterator[Dataset]:
+    r"""Iterates over all view codes in an input."""
+    view_code_sequence = (get_value(dcm, Tag.ViewCodeSequence, []) or []) if isinstance(dcm, Dataset) else dcm
```

Comment:
Is the `or []` redundant since `[]` is already passed as the default to `get_value`?

Reply:
This was in the following function as well.

Reply:
I believe this was to deal with Tag.ViewCodeSequence being present with a value of `None`, so it is necessary.

Reply:
Ah, that makes sense. Should we consider handling that situation inside of `get_value`, or is there a place where we would want `None` returned?

Reply:
This is probably a rare situation where we explicitly need the output to be a sequence. I would be inclined to leave `get_value` as is, unless we start encountering this type of situation more frequently with a clear pattern to how the output should be handled.

Reply:
Sounds good.